tests/list-to-attrs-nullptr.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
```

Comment:
Rather than adding a brand new toplevel test (which does cost a bit as initializing a test is slightly expensive), I think you can

- Add this to `tests/lang/eval-fail-list-to-attrs.nix`
- Change https://github.com/NixOS/nix/blob/master/tests/lang.sh#L35 to `if ! expect 1 nix-instantiate --eval lang/$i.nix; then` (I checked, all the other tests do return `1` as they should)

Also: Can you add a reference to the issue in the test file, so that we know why it’s here?